test/compression.js

```diff
@@ -676,6 +676,76 @@ describe('compression()', function () {
       .end()
     })
   })
+
+  describe('when callbacks are used', function () {
+    it('should call the passed callbacks in the order passed when compressing', function (done) {
+      var hasCallbacks = false
+      var callbackOutput = []
+      var server = createServer(null, function (req, res) {
+        hasCallbacks = (res._write.length === 3 && res._end.length === 3)
+        res.setHeader('Content-Type', 'text/plain')
+        res.write('Hello', null, function () {
+          callbackOutput.push(0)
+        })
+        res.write(' World', null, function () {
+          callbackOutput.push(1)
+        })
+        res.end(null, null, function () {
+          callbackOutput.push(2)
+        })
+      })
+
+      request(server)
+      .get('/')
+      .set('Accept-Encoding', 'gzip')
+      .expect('Content-Encoding', 'gzip')
+      .end(function (err) {
+        if (err) {
+          throw new Error(err)
```

Comment:
You don't want to wrap an error ina new error, because it looses the original stack trace. It is good practice to test your error handling code to see that it works, just like the normal code. This should simply be passing err to the done function.